groups/data-reconciliation/main.tf

```diff
@@ -0,0 +1,122 @@
+    region = var.aws_region
+  }
+}
+
+data "terraform_remote_state" "services-stack-configs" {
+  backend = "s3"
+  config = {
+    bucket = var.aws_bucket # aws-common-infrastructure-terraform repo uses the same remote state bucket
+    key    = "aws-common-infrastructure-terraform/common-${var.aws_region}/services-stack-configs.tfstate"
+    region = var.aws_region
+  }
+}
+
+data "vault_generic_secret" "secrets" {
+  path = "applications/${var.aws_profile}/${var.environment}/${local.stack_fullname}"
+}
+
+resource "aws_ssm_parameter" "secret_parameters" {
+  for_each = data.vault_generic_secret.secrets.data
+    name = "/${local.stack_fullname}/${each.key}"
+    key_id = data.terraform_remote_state.services-stack-configs.outputs.services_stack_configs_kms_key_id
+    description = each.key
+    type = "SecureString"
+    overwrite = "true"
+    value = each.value
+}
+
+locals {
+  ecs_task_config = merge({
+    aws_region = var.aws_region
+    service_name = local.service_name
+    name_prefix = local.name_prefix
+    jdbc_driver = var.jdbc_driver
+    company_count_crontab = var.company_count_crontab
+    company_count_oracle = var.company_count_oracle
+    company_count_mongodb = var.company_count_mongodb
+    company_collection_crontab = var.company_collection_crontab
+    company_collection_oracle = var.company_collection_oracle
+    company_collection_mongodb = var.company_collection_mongodb
+    endpoint_output = var.endpoint_output
+    docker_registry = var.docker_registry
+    release_version = var.release_version
+  }, aws_ssm_parameter.secret_parameters)
+}
+
+module "ecs-cluster" {
+  source = "git::git@github.com:companieshouse/terraform-library-ecs-cluster.git?ref=1.1.1"
+  stack_name = local.stack_name
+  name_prefix = local.name_prefix
+  environment = var.environment
+  vpc_id = data.terraform_remote_state.networks.outputs.vpc_id
+  subnet_ids = data.terraform_remote_state.networks.outputs.application_ids
+  ec2_key_pair_name = var.ec2_key_pair_name
+  ec2_instance_type = var.ec2_instance_type
+  ec2_image_id = var.ec2_image_id
+  asg_max_instance_count = 1
+  asg_min_instance_count = 1
+  asg_desired_instance_count = 1
+}
+
+resource "aws_security_group" "data-reconciliation-security-group" {
+  name = local.security_group_name
+  description = "Security group for data reconciliation"
+  egress {
+    from_port = 0
+    to_port = 0
+    protocol = "-1"
+    cidr_blocks = [
+      "0.0.0.0/0"
+    ]
+  }
+  tags = {
+    Name = local.security_group_name
+    Environment = var.environment
+    Service = local.stack_name
+  }
+}
+
+resource "aws_ecs_task_definition" "data-reconciliation-task-definition" {
+  family = local.ecs_service_name
+  execution_role_arn = module.ecs-cluster.ecs_task_execution_role_arn
+  container_definitions = templatefile(local.task_config_file, local.ecs_task_config)
+}
+
+resource "aws_ecs_service" "data-reconciliation-ecs-service" {
```

Comment:
This structure is fine for a single ECS Service/Task but as we add more services/tasks configured to collect different metrics we'll need to refactor these elements into a separate module as we do in other ECS terraform repos. 

It might save confusion and a bit of refactor work later on if this is done now but if the team want to pick up the restructure work when adding the second ECS service/task thats fine.

Reply:
Hi Billy - we will refactor our terraform code into separate modules at a later date after we have brought in a task to rework it into separate modules.